### PR TITLE
BlogCard and BlogCardList

### PR DESCRIPTION
**Pull request by jacorea**
Added BlogCard and BlogCardList. Also updated agent.js to include API calls to get FeaturedBlogs as well as well as all blogs to display on BlogCardList component.

### CODE REVIEW

**Comment by vickyadebona** [comment]
@jacorea seems like you have a few merge conflicts

File: frontend/src/pages/Home/HomePageBlogSection.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      try {
+        const blogCardData = await FeaturedBlogposts.getFeatured().then(
+          response => response
+        );
+        setIsLoading(false);
+        setFeaturedBlogs(blogCardData);
+      } catch (error) {
+        console.log(error);
+      }
+    };
```
Comment (dbtung) [inline frontend/src/pages/Home/HomePageBlogSection.jsx:26]: C

File: frontend/src/components/BlogCardList/BlogCardList.jsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    .slice(0, wordLimit)
+    .join(' ');
+  return `${description}`;
+};
+
+//this function deterimines if media_url is an image or video
+const isVideoOrImage = urlStr => {
+  //need to check end of string to see if contains jpeg, jpg, png,
+  // let  extensionsArr = [ "jpeg", "jpg", "gif", "png" ];
+  //rather than using regex can try stripping off last part of url, and seeing if that bit contains any of the image extenstions
```
Comment (dbtung) [inline frontend/src/components/BlogCardList/BlogCardList.jsx:53]: Should just simplify this comment to be `// Check the end of the string to see if it contians jpeg, jpg, or png.`.

File: frontend/src/components/BlogCard/BlogCard.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        active={true}
+        aspectRatio="4:3"
+        placeholder="/images/image-16by9.png"
+        source="youtube"
+      />
+    );
+  }
+};
+
+const BlogCard = ({ blogImgurl, blogTitle, blogDescription, blogVideo }) => {
```
Comment (dbtung) [inline frontend/src/components/BlogCard/BlogCard.jsx:70]: nit: `blogImgurl` -> 'blogImgUrl`

File: frontend/src/components/BlogCardList/BlogCardList.jsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  cardCount
+}) => {
+  const cards = !cardInfo
+    ? null
+    : cardInfo.slice(0, !cardCount ? cardInfo.length : cardCount).map(card => {
+        return (
+          <Grid.Column computer={5} mobile={16} tablet={7} key={card.id}>
+            <BlogCard
+              blogImgurl={isVideoOrImage(card.blogpost.media_url)}
+              blogVideo={removeHtmlTags(card.blogpost.media_url)}
```
Comment (dbtung) [inline frontend/src/components/BlogCardList/BlogCardList.jsx:80]: Now that I look at this some more, it's a bit strange to me that you have some branching condition here for figuring out things about whether the URL is an image or a video, and then you do other work in `BlogCard` - I would recommend moving all of the responsibility for worrying whether the URL is an image or a video into `BlogCard` and just pass the mediaURL.

File: frontend/src/components/BlogCardList/BlogCardList.jsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  let isImage = urlStr.match(/\.(jpeg|jpg|gif|png)$/) != null;
+  return isImage ? alteredImage : null;
+};
+
+/* INFO
+Props: blogHeader, linkText, cardInfo, wordCount, cardCount
+to use the component can nest it inside your section component else it takes up full width of page
+Can have it display the Topic header if you want or you could make your own if don't pass a blogHeader to your CardList
+wordCount and cardCount limit the number of cards displayed, by default it displays all the cards in cardInfo array that is passed
+card info is your array with your desired */
```
Comment (dbtung) [inline frontend/src/components/BlogCardList/BlogCardList.jsx:64]: Same comments about the commenting here.

File: frontend/src/pages/Blog/Blog.jsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  line-height: 118.5%;
+  text-align: center;
+  color: ${theme.colors.white};
+  width: 330px;
+  height: 76px;
+  margin: 25px;
+  :hover {
+    background: ${theme.colors.lightPurple};
+  }
+`;
```
Comment (dbtung) [inline frontend/src/pages/Blog/Blog.jsx:27]: Use theme constants.

File: frontend/src/pages/Blog/BlogTopicsSection.jsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  
+  :hover {
+    color: ${theme.colors.lightPurple};
+  }
+
+  :active {
+    color: #341A93;
+    text-decoration: underline;
+  }
+`;
```
Comment (dbtung) [inline frontend/src/pages/Blog/BlogTopicsSection.jsx:24]: Use theme constants.

File: frontend/src/pages/Blog/BlogTopicsSection.jsx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+const BlogTopicsSection = () => {
+  const [blogTopicName, setBlogTopicNames] = useState([]);
+  const [isBlogTopicNameLoading, setBlogTopicNameLoading] = useState(true);
```
Comment (dbtung) [inline frontend/src/pages/Blog/BlogTopicsSection.jsx:28]: This is a super-nit, but try to keep it consistent whether it should be plural or singular. Seems like here it should be plural.

File: frontend/src/pages/Blog/BlogTopicsSection.jsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+                <Grid.Column>
+                  <StyledLink to="/">view all</StyledLink>
+                </Grid.Column>
+              </List.Item>
+            </List>
+          </Grid.Column>
+        </Grid.Row>
+        <Grid.Row width={8}>{blogTopics}</Grid.Row>
+      </Grid>
+    </Segment>
```
Comment (dbtung) [inline frontend/src/pages/Blog/BlogTopicsSection.jsx:76]: Consider using a Loading component to wrap it all up.

File: frontend/src/pages/Blog/FeaturedBlogSection.jsx [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+  }
+`;
+
+const WebinarLink = Styled(StyledLink)`
+  color: ${theme.colors.lightPurple};
+  text-decoration: underline;
+  :active {
+    color: ${theme.colors.purple};
+  }
+`;
```
Comment (dbtung) [inline frontend/src/pages/Blog/FeaturedBlogSection.jsx:40]: Just combine it all into WebinarLink.

File: frontend/src/pages/Blog/FeaturedBlogSection.jsx [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+import { Grid, Segment, Embed } from 'semantic-ui-react';
+import Styled from 'styled-components';
+import { Link } from 'react-router-dom';
+import theme from '../../styles/theme';
+
+/* TODO
+1)remove html tags for content
+2)Create a function that determines if media_url is for image or video.
+3)If video then use the Embed component to display the video.
+4)turn iso date-time format into date format to return Month, date and Year to display
```
Comment (dbtung) [inline frontend/src/pages/Blog/FeaturedBlogSection.jsx:12]: For generating readable dates, I like using `moment.js`

File: frontend/src/pages/Blog/FeaturedBlogSection.jsx [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+    blogInfo();
+  }, []);
+
+  const featuredBlogInfo = bloglist => {
+    return isFeaturedBlogLoading
+      ? null
+      : bloglist.slice(0, 1).map(featured => {
+          return (
+            <Segment key={featured.id}>
+              {console.log(featured)}
```
Comment (dbtung) [inline frontend/src/pages/Blog/FeaturedBlogSection.jsx:67]: Get rid of this?

File: frontend/src/pages/Blog/FeaturedBlogSection.jsx [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+              <Grid doubling stackable>
+                <Grid.Column>
+                  <h3>Featured</h3>
+                  {console.log(featured.blogpost.media_url)}
```
Comment (dbtung) [inline frontend/src/pages/Blog/FeaturedBlogSection.jsx:71]: Here too?

**Pull request closed without merging**